Fixes #17867 - Show existing products on discovery create

dLobatog opened the pull request:
The create products after discovery page does not use the styles
described on
http://www.patternfly.org/pattern-library/forms-and-controls/forms/#_

Another problem it has is that even when there are no products
available, it shows you an option 'existing product' but clicking on it
does nothing at all

![screenshot from 2016-12-28 11-46-53](https://cloud.githubusercontent.com/assets/598891/21521372/8e8ac56a-ccfc-11e6-8813-df1f7c67ef42.png)

ehelms commented:
Can oyu fix the HTML indentation to two spaces?

ehelms commented:
I'd also consult with @waldenraines whether this should adopt to his changes around using horizontal forms.

I tend to like having options around even if they are empty, otherwise I think users tend not to know they even exist. So if we hide the ability to use an existing product just because there aren't currently, users may or may not learn they could use an existing product (or might ask themselves, why isn't my product showing up -- do I not have permissions? Is there an error ?)

dLobatog commented:
Fixed the indentation and now the form uses bst-form-group from Bastion (like most other Katello forms). I removed the `if condition` about "exisiting products" as that seems broken to me, no matter if I have custom content or not, it never shows up.

Errors didn't display properly prior to this change, I modified the controller so they do now.

dLobatog commented:
![screenshot from 2016-12-28 18-08-38](https://cloud.githubusercontent.com/assets/598891/21527329/e3b20520-cd2a-11e6-84cb-a5429d7cabd8.png)

waldenraines commented on engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/views/discovery-create.html:
```diff
@@ -1,91 +1,89 @@
 <span page-title>{{ 'Create Repositories' | translate }}</span>
 
-<form name="productForm" class="col-sm-5" ng-disabled="creating()" novalidate role="form">
+<form name="productForm" class="col-sm-5 form-horizontal" ng-disabled="creating()" novalidate role="form">
```
We are moving away from using `.form-horizontal`.  Do you mind changing this to use the normal bs3 form?

waldenraines commented on engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/views/discovery-create.html:
```diff
@@ -95,48 +93,39 @@ <h4 ng-show="discovery.selected.length > 0" translate>Selected URLs:</h4>
+      ng-disabled="creating() || !requiredFieldsEnabled()"
+      translate
+      class="btn btn-primary">
+      <i class="pficon-add-circle-o inline-icon"></i>	Create
```
Can you remove this icon?  We are moving towards not having icons in the buttons.

dLobatog replied:
No problem - I added the icon because of https://groups.google.com/forum/?fromgroups#!topic/foreman-dev/jy2Ytas4SRw , @rohoover any idea what to do?

waldenraines replied:
Ah okay, I just felt it was superfluous but I don't have too strong of an opinion about it.

waldenraines replied:
I don't want to hold this PR up on one icon.

dLobatog commented:
Repushed:

* I removed form-horizontal
* The controller now uses isUndefined to comply with eslint rules

waldenraines commented on engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/views/discovery-create.html:
```diff
@@ -3,89 +3,87 @@
-                ng-options="product.id as product.name for product in products">
-        </select>
+
+    <div class="clearfix">
```
Why is this needed?

dLobatog commented:
@waldenraines No reason at all for the various 'clearfixes' , thanks for catching that. Notice the list of existing products does *not* show up either with or without this change, I'm trying to debug the form controller but still I'm not clear what's going on (Angular newbie here)

dLobatog commented:
@waldenraines Fixed the issue with showing existing products: ng-hide couldn't work because it was checking on newProduct === `true` (boolean) but `"true"` is a string in the discovery-form controller. Those were all the changes I wanted to make. Thanks for reviewing

dLobatog commented on engines/bastion_katello/app/assets/javascripts/bastion_katello/products/discovery/views/discovery-create.html:
```diff
@@ -3,89 +3,76 @@
-    <div class="form-group">
-      <div class="col-md-5" ng-hide="createRepoChoices.newProduct">
+
+    <div ng-hide="createRepoChoices.newProduct === 'true'">
```
This didn't work before because newProduct is a string, not a boolean

waldenraines replied:
Is there a way to cast it to a boolean?  This seems rather fragile to me.

dLobatog replied:
@waldenraines The problem is that `newProduct` comes from the HTML tag value at lines 14 and 23. Those are sent as strings to angular and I don't think they can be sent as booleans. 

To auto parse them as booleans I could write a directive like http://jsfiddle.net/zk99k4fg/1/ but I think it's overkill in this case (what do we gain from checking the string vs a boolean in this case?)

dLobatog commented:
@waldenraines Replied inline to the issue of `newProduct` being a 'true/false' string vs boolean.

dLobatog commented:
@waldenraines Anything I'm missing here?

waldenraines commented:
> @waldenraines Anything I'm missing here?

No, sorry I missed this, approved now.

jlsherrill commented:
Thanks @dLobatog !
